specs/src/tx_client.md

```diff
@@ -0,0 +1,273 @@
+```
+
+- **Path**: Direct processing, no worker queue
+- **Input**: Cosmos SDK messages
+- **Process**: Transaction building -> Broadcast -> Confirmation
+- **Usage**: Should only work with sequential submission
+
+#### SubmitPayForBlob
+
+```go
+func (client *TxClient) SubmitPayForBlob(ctx context.Context, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error)
+```
+
+- **Path**: Direct processing with default account, no worker queue
+- **Input**: Celestia blobs
+- **Process**: PayForBlob creation -> Broadcast -> Confirmation
+- **Usage**: Only for sequential submissions using the default account
+
+#### SubmitPayForBlobToQueue
+
+```go
+func (client *TxClient) SubmitPayForBlobToQueue(ctx context.Context, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error)
+```
+
+- **Path**: Uses worker queue for parallel processing
+- **Input**: Celestia blobs
+- **Process**: Job queued -> Worker assignment -> PayForBlob creation -> Broadcast -> Confirmation
+- **Usage**: Works with sequential processing with a single worker or parallel processing with multiple. If you only create a single worker it will be sequential processing and ordered. If there are multiple workers then it will spin up as many accounts as the workers and submits in parallel.
+
+#### SubmitPayForBlobWithAccount
+
+```go
+func (client *TxClient) SubmitPayForBlobWithAccount(ctx context.Context, accountName string, blobs []*share.Blob, opts ...TxOption) (*TxResponse, error)
+```
+
+- **Path**: Bypasses worker queue, uses specified account directly
+- **Input**: Celestia blobs + account name
+- **Process**: Direct account usage -> PayForBlob creation -> Broadcast -> Confirmation
+- **Usage**: This bypasses having to spin up workers at all and can be used with a signle account for sequential submissions
+
+#### QueueBlob
+
+```go
+func (client *TxClient) QueueBlob(ctx context.Context, resultC chan SubmissionResult, blobs []*share.Blob, opts ...TxOption)
+```
+
+- **Path**: Direct job submission to worker queue
+- **Input**: Context, result channel, Celestia blobs, and transaction options
+- **Process**: Job queued directly -> Worker assignment -> Processing
+- **Usage**: This method adds a job into the queue directly, bypassing the normal submission flow
+
+### Submission API Summary
+
+All submission APIs ultimately delegate into the same broadcast and confirmation methods, they differ only in account selection, tx types and whether jobs go through the worker queue.
+
+## Broadcast
+
+Broadcasting steps are universal across SDK transactions and blob transactions.
+
+**Process:**
+
+1. **Build & Sign**: Transaction is built and signed using the Signer
+2. **Gas Estimation**:
+   - If no gas provided, simulate with minimal fee
+   - Call gas estimation service to determine gas usage and price
+   - If estimation fails due to a sequence mismatch, update local sequence to expected and retry
```

Comment:
There's still some debate around whether we should actually be doing this (updating local sequence with what consensus node hands back). I'm not sure if we should specify this behaviour rn 

cc @cmwaters

Reply:
I think it's very confusing when you get sequence mismatch during simulation so until this behaviour is resolved i think we should mention it as this is the only way to bypass sequence mismatch during estimation

Reply:
just to note, on rust side we went with re-querying the account instead of matching a regex. Probably less responsive / slower approach, but felt more correct